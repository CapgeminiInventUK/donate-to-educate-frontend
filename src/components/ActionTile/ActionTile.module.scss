@use '@assets/global.scss' as g;

@mixin icons($background-color) {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  display: flex;
  text-align: center;
  justify-content: center;
  align-items: center;
  background-color: $background-color;

  @include g.sm-breakpoint {
    margin: 0 0 5px 20px;
    min-width: 50px;
  }
}

@mixin tile($border-top, $background, $color) {
  display: flex;
  flex: 1;
  block-size: fit-content;
  padding: 30px 40px;
  gap: 20px;
  border-radius: 10px;
  border-top: $border-top 8px solid;
  background: $background;
  color: $color;

  h3 {
    padding: 0;
    margin: 0;
  }

  .tileText {
    position: relative;
    padding-left: 20px;

    p {
      @include g.md-breakpoint {
        font-size: 16px;
      }
    }

    .button {
      position: relative;
      white-space: nowrap;
    }
  }

  @include g.md-breakpoint {
    gap: 10px;
  }

  @include g.sm-breakpoint {
    flex-direction: column;
    padding-bottom: 40px;
    .button {
      height: 35px;
    }
  }
}

.grey {
  @include tile(g.$primary-light-blue, g.$about-grey, g.$dark-blue);

  .greyIcon {
    @include icons(g.$primary-light-blue);
  }
}

.lightBlue {
  @include tile(g.$primary-mid-blue, g.$primary-light-blue, g.$dark-blue);

  .lightBlueIcon {
    @include icons(g.$primary-mid-blue);
  }
}

.midBlue {
  @include tile(g.$dark-blue, g.$primary-mid-blue, g.$primary-white);

  .midBlueIcon {
    @include icons(g.$primary-dark-blue);
  }
}

.darkBlue {
  @include tile(g.$primary-light-blue, g.$primary-dark-blue, g.$primary-white);

  .darkBlueIcon {
    @include icons(g.$primary-light-blue);
  }
}
