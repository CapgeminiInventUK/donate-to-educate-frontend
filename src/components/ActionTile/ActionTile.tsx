import { FC } from 'react';
import styles from './ActionTile.module.scss';
import { ActionTileProps } from '@/types/props';
import FormButton from '../FormButton/FormButton';

const ActionTile: FC<ActionTileProps> = ({
  icon,
  heading,
  subheading,
  buttonText,
  theme,
  isAdmin = false,
  onClick,
}) => {
  return (
    <div aria-label="action-tile" className={styles[theme]}>
      <div aria-label="action-tile-icon" className={`${styles[theme + 'Icon']}`}>
        {icon}
      </div>
      <div className={styles.tileText}>
        <h3>{heading}</h3>
        <p>{subheading}</p>
        <FormButton
          className={styles.button}
          theme={
            !isAdmin ? 'formButtonGreen' : theme === 'grey' ? 'formButtonMidBlue' : 'formButtonGrey'
          }
          text={buttonText}
          onClick={onClick}
          useArrow={!buttonText.includes('Edit')}
          ariaLabel={`${buttonText} button`}
        />
      </div>
    </div>
  );
};

export default ActionTile;
