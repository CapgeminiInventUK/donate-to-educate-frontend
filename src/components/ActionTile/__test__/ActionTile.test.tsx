import type { TileThemes } from '@/types/props';
import { render } from '@testing-library/react';
import buttonStyles from '../../FormButton/FormButton.module.scss';
import ActionTile from '../ActionTile';
import styles from '../ActionTile.module.scss';

const props = {
  icon: <></>,
  heading: 'Heading',
  subheading: 'Subheading',
  buttonText: 'Button text',
  theme: 'lightBlue' as TileThemes,
  isAdmin: false,
  onClick: vi.fn(),
};

afterEach(() => {
  props.buttonText = 'Button text';
  props.theme = 'lightBlue';
  props.isAdmin = false;
});

describe('Action tile', () => {
  it('should render component', () => {
    const screen = render(<ActionTile {...props} />);
    expect(screen).toMatchSnapshot();
  });

  it('should alter classnames depending on theme', () => {
    props.theme = 'grey';
    const { getByLabelText, getByRole } = render(<ActionTile {...props} />);
    expect(getByLabelText('action-tile')).toHaveClass(styles.grey);
    expect(getByLabelText('action-tile-icon')).toHaveClass(styles.greyIcon);
    expect(getByRole('button')).toHaveClass(buttonStyles.formButtonGreen);
  });

  it('should alter button classname depending on theme and isAdmin', () => {
    props.isAdmin = true;
    const { getByRole } = render(<ActionTile {...props} />);
    expect(getByRole('button')).toHaveClass(buttonStyles.formButtonGrey);
  });

  it('should alter button classname depending on theme', () => {
    props.theme = 'grey';
    props.isAdmin = true;
    const { getByRole } = render(<ActionTile {...props} />);
    expect(getByRole('button')).toHaveClass(buttonStyles.formButtonMidBlue);
  });
});
