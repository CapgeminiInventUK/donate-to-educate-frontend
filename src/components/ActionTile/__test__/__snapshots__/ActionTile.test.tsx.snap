// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Action tile > should render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="action-tile"
        class="_lightBlue_932d70"
      >
        <div
          aria-label="action-tile-icon"
          class="_lightBlueIcon_932d70"
        />
        <div
          class="_tileText_932d70"
        >
          <h3>
            Heading
          </h3>
          <p>
            Subheading
          </p>
          <button
            aria-label="Request button"
            class="_formButtonGreen_a4d411  _button_932d70"
          >
            <span
              class="_text_a4d411"
            >
              Button text
            </span>
            <div>
              <img
                alt="Right arrow"
                height="30"
                src="/src/assets/icons/form-button-right-arrow-white.svg"
                width="30"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="action-tile"
      class="_lightBlue_932d70"
    >
      <div
        aria-label="action-tile-icon"
        class="_lightBlueIcon_932d70"
      />
      <div
        class="_tileText_932d70"
      >
        <h3>
          Heading
        </h3>
        <p>
          Subheading
        </p>
        <button
          aria-label="Request button"
          class="_formButtonGreen_a4d411  _button_932d70"
        >
          <span
            class="_text_a4d411"
          >
            Button text
          </span>
          <div>
            <img
              alt="Right arrow"
              height="30"
              src="/src/assets/icons/form-button-right-arrow-white.svg"
              width="30"
            />
          </div>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
